### PR TITLE
Evv446 r1 tests

### PR DESCRIPTION
**Pull request by eileenvanheerde**
Wrote some tests for the R1 tests cases on the test matrix. They kind of rely on the parse() function being fixed though, so quite a few are currently commented out. I also need to refactor them a bit to try and minimize the repetition as much as I can. 

I'm also having some trouble getting the snapshot to work for the multiple date format tests. They pass when I run them on my own machine, but they fail when I push them. I think it has something to do with momentjs and how the "Date_Num" is produced.

### CODE REVIEW

File: src/__tests__/ParserComponent.test.tsx [hunk 1]
```diff
@@ -61,22 +62,214 @@ describe('Csv FileEvents processed correctly', () => {
...
+          'multiDateTest.csv',
+          {type: '.csv,text/csv'},
+      );
+      const fileEvent = {target: {files: [multiDateFile]}};
+
+      // no error should be thrown!!!
+      try {
+        expect(await comp.instance().parse(fileEvent)).toBe(undefined);
+      } catch (error) {
+        fail(); // fail if error thrown
```
Comment (braunsonm) [inline src/__tests__/ParserComponent.test.tsx]: Shouldn't this just fail if an exception is thrown anyways?

File: src/__tests__/ParserComponent.test.tsx [hunk 2]
```diff
@@ -61,22 +62,214 @@ describe('Csv FileEvents processed correctly', () => {
...
+        {type: '.csv,text/csv'},);
+    const comp: any = mount(<ParserComponent
+      {...props}
+      onChange={onChangeMock}
+    />);
+    const fileEvent = {target: {files: [noDateFile]}};
+    try {
+      expect(
+          await comp.instance().parse(fileEvent)
+      ).toThrowError('The file uploaded has no dates.');
```
Comment (braunsonm) [inline src/__tests__/ParserComponent.test.tsx]: Nice, this is how I'd expect the error to be checked. However are you sure you need to wrap it in a try catch?

File: src/__tests__/ParserComponent.test.tsx [hunk 3]
```diff
@@ -61,22 +62,214 @@ describe('Csv FileEvents processed correctly', () => {
...
+
+      // no error should be thrown!!!
+      try {
+        expect(await comp.instance().parse(fileEvent)).toBe(undefined);
+      } catch (error) {
+        fail(); // fail if error thrown
+      }
+      // data should be updated to contain csv info
+      compData = comp.state('data');
+      // Check that the object contains all the data from the csv
```
Comment (braunsonm) [inline src/__tests__/ParserComponent.test.tsx]: But it doesn't really check that right? It just sees that it turned into "some number"? What if it incorrectly parses the dates?

**Comment by braunsonm** [comment]
Also I can take a look at the build issue, I'm having trouble viewing the logs right now of the output :(

File: src/__tests__/ParserComponent.test.tsx [hunk 4]
```diff
@@ -61,22 +62,214 @@ describe('Csv FileEvents processed correctly', () => {
...
+
+    // make sure mock is cleared
+    beforeEach(() => {
+      onChangeMock.mockClear();
+      compData = [];
+    });
+
+    afterEach(() => {
+      compData = comp.state('data');
+      // expect(onChangeMock).not.toHaveBeenCalled();
```
[the remark this answers is not in the document]
Comment (aaz713) [reply src/__tests__/ParserComponent.test.tsx]: i'm gonna assume it has something to do with the parse() function being broken but i'm not completely sure if that's  the case

File: src/__tests__/ParserComponent.test.tsx [hunk 5]
```diff
@@ -61,22 +62,214 @@ describe('Csv FileEvents processed correctly', () => {
...
+      expect(compData.length).toBe(0);
+    });
+
+    it('.pdf rejected', async () => {
+      const pdfTestFile: File = new File(
+          ['test'],
+          'test.pdf',
+          {type: '.pdf,application/pdf'},
```
[the remark this answers is not in the document]
Comment (kevinwish) [reply src/__tests__/ParserComponent.test.tsx]: Do you mean such as naming a .pdf a .csv file?

[the remark this answers is not in the document]
Comment (kevinwish) [reply src/__tests__/ParserComponent.test.tsx]: > 
> 
> What if you were to lie about the type?

Do you mean such as naming a .pdf a .csv file?

[the remark this answers is not in the document]
Comment (braunsonm) [reply src/__tests__/ParserComponent.test.tsx]: Yes and/or submitting a `.pdf` but changing the type variable to something that makes it look like a CSV

**Pull request merged**